x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/plugin/RestSqlQueryAction.java

```diff
@@ -95,7 +95,7 @@ protected RestChannelConsumer prepareRequest(RestRequest request, NodeClient cli
             return channel -> client.execute(SqlQueryAction.INSTANCE, sqlRequest, new RestResponseListener<SqlQueryResponse>(channel) {
                 @Override
                 public RestResponse buildResponse(SqlQueryResponse response) throws Exception {
-                    XContentBuilder builder = XContentBuilder.builder(xContentType.xContent());
+                    XContentBuilder builder = channel.newBuilder(request.getXContentType(), xContentType, false);
```

Comment:
You can as well set filtering to `true` for consistency. I don't see how this can be very useful at the moment in this particular case besides turning off rows, but if we will add some other stuff there, it can help. It will basically allow users to do something like this:
```
POST _sql?filter_path=rows
{
  "query": "SELECT 1"
}
```